### PR TITLE
DT-2966: 🐛 Fix swagger docs

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt [hunk 1]
```diff
@@ -8,11 +8,9 @@ import org.springframework.security.config.annotation.web.builders.HttpSecurity
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
 import org.springframework.security.config.http.SessionCreationPolicy
-import springfox.documentation.swagger2.annotations.EnableSwagger2
 
 @Configuration
 @EnableWebSecurity
-@EnableSwagger2
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt:15]: only needed for v2

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt [hunk 2]
```diff
@@ -37,9 +35,8 @@ open class ResourceServerConfiguration : WebSecurityConfigurerAdapter() {
         auth.antMatchers(
           "/webjars/**", "/favicon.ico", "/csrf",
           "/health/**", "/info", "/h2-console/**",
-          "/v2/api-docs",
-          "/swagger-ui.html", "/swagger-resources", "/swagger-resources/configuration/ui",
-          "/swagger-resources/configuration/security"
+          "/v3/api-docs", "/swagger-ui/**",
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/ResourceServerConfiguration.kt:38]: switch to v3

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -69,6 +69,11 @@ spring:
         provider:
           hmpps-auth:
             token-uri: ${api.base.url.oauth}/oauth/token
+
+  mvc:
+    pathmatch:
+      matching-strategy: ant_path_matcher
```
Comment (petergphillips) [inline src/main/resources/application.yml:75]: magic line to get docs working

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/repository/OpenApiDocsTest.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      .exchange()
+      .expectStatus().isOk
+      .expectBody().jsonPath("info.version").isEqualTo(DateTimeFormatter.ISO_DATE.format(LocalDate.now()))
+  }
+
+  @Test
+  fun `the open api json is nearly valid and contains documentation`() {
+    val result = OpenAPIV3Parser().readLocation("http://localhost:$port/v3/api-docs", null, null)
+    assertThat(result.openAPI.paths).isNotEmpty
+    assertThat(result.messages).hasSizeLessThan(2)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/repository/OpenApiDocsTest.kt:50]: there is still one error here to do with response `Page<NhsPrisoner>` but ran out of time to get it working

**Review by neilmendum — APPROVED** [review]

**Pull request merged**